### ISSUE TITLE
Fault Tree Analysis has an infinite loop / gives an unhandled event loop exception

### ISSUE THREAD

**Issue by sprocter** [osate/osate2#2115]
## Summary

I'm trying to compute a fault tree for part of an aircraft's wheel braking system, but getting an error that there's an infinite loop / stack overflow occurring.

## Expected and Current Behavior

**Expected:** OSATE should generate and display a fault tree.

**Actual:** A popup box titled "Problem Occurred" appears, and says: 

> Unhandled event loop exception
> java.lang.StackOverflowError

The terminal fills up with these lines:

```
	at org.osate.aadl2.errormodel.faulttree.generation.FTAGenerator.optimizeGates(FTAGenerator.java:543)
	at org.osate.aadl2.errormodel.faulttree.generation.FTAGenerator.optimizeGates(FTAGenerator.java:543)
	at org.osate.aadl2.errormodel.faulttree.generation.FTAGenerator.optimizeGates(FTAGenerator.java:543)
```

## Steps to Reproduce

1. Load the "GTSE-Examples" project from the [GATSE repo](https://github.com/osate/osate2-gtse/tree/master/GTSE-Examples/ARP4761) into OSATE
2. Instantiate the `bscu.generic` system implementation, which is located in `GTSE-Examples/ARP4761/advanced/implementation/bscu.aadl`
3. Run the Fault Tree Analysis tool on the model with the following (default) settings:
![FTA Exception Screenshot](https://user-images.githubusercontent.com/1319948/70923191-78a8d300-1ff5-11ea-959e-a0489fee1106.png)

## Environment

* **OSATE Version**: OSATE is built from source. I'm on the master branch; the most recent commit is fafd82c38837a4749b778656746156e3a390c4ab.
* **Operating System**: Mac OS 10.14.6

**Comment by lwrage** [comment]
The exception happens in OSATE 2.5.1, already, so this is not a newly introduced bug.

**Comment by reteprelief** [comment]
Addressed by fix for issue #2123 in pull request #2135
